blog/2020-04-07-ModernJavaConstraintsNoDRL.adoc

```diff
@@ -0,0 +1,153 @@
+= Modern java constraints (no Drools Rule Language) - ConstraintStreams
+:page-interpolate: true
+:awestruct-author: ge0ffrey
+:awestruct-layout: blogPostBase
+:awestruct-tags: [feature, coding, constraint]
+:awestruct-share_image_filename: constraintStreamGroupBy.png
+
+Traditionally, to scale out with https://www.optaplanner.org/[OptaPlanner],
+you had to learn DRL. _No more._
+With the new _ConstraintsStreams_ API, inspired Java 8 Streams and SQL,
+you can now *write your constraints in Java* (or Kotlin or Scala)
+and still benefit from incremental calculation.
+
+Underneath, ConstraintsStreams (CS) still use the powerful https://www.drools.org/[Drools] engine.
+We also still fully support score DRL's too. They are _not_ deprecated.
+
+Let's start with an example.
+In nurse rostering, to avoid assigning shifts to employee `Ann`,
+you would write this constraint in DRL:
+
+[source,drl]
+----
+rule "Don't assign Ann"
+    when
+        Shift(getEmployee().getName() == "Ann")
```

Comment:
Actually, in DRL, you'd probably just write `Shift(employee.name == "Ann")`.

Reply:
Yes, but I discuss that later, so here I just used the java form.
Actually, "employee.name" doesn't work well, as employee is null when an entity is initialized and it throws wierd NPE's.

Reply:
Leaving this as is